starfish/imagestack/physical_coordiante_calculator.py

```diff
@@ -0,0 +1,114 @@
+from typing import Mapping, Tuple, Union
+
+import xarray as xr
+
+from starfish.types import Coordinates, Indices, PhysicalCoordinateTypes
+
+
+def calc_new_physical_coords(image_stack, indexers) -> xr.DataArray:
+    """Calculates the resulting coordinates array from indexing on each dimension in indexers
+
+    Parameters
+    ----------
+    indexers : a dictionary of dim:index where index is the value
+    or range to index the dimension
+
+    """
+    new_coords = image_stack._coordinates.copy()
+    # index by R, CH, V
+    key = {'r': indexers[Indices.ROUND.value],
+           'c': indexers[Indices.CH.value],
+           'z': indexers[Indices.Z.value]}
+    new_coords = image_stack.index_keep_dimensions(new_coords, key)
```

Comment:
`ImageStack.sel` itself calls `index_keep_dimensions`, and then it also calls `calc_new_physical_coords`, which calls `index_keep_dimensions` again.  This doesn't seem right.

I propose that `calc_new_physical_coords` takes:
1. named shape of the original 5D tensor
2. named physical coordinates of the original 5D tensor
3. named indexers

From that, you should be able to calculate the coordinates of the indexed array.

Reply:
Well I'm calling index keep dimensions on two different xarrays. One is Imagestack.xarray in sel just indexing by whatever is passed in and the second is the coordinates xarray that should also be indexed by the same R, CH, V values before then calculating the actual coordinate values. Which part doesn't seem right? Both xarrays need to be passed into index_keep_dimensions at some point, I guess I could do both in the sel method but my thinking was that calc_new_physical_coords was doing everything to "calculate a new physical coords xarray"

Reply:
Ahh, I see.  I totally misread the code.  I think that furthers the argument for moving `index_keep_coordinates` out of `ImageStack`.

You're totally right that we need to slice both the image data and the coordinates.  However, even after better understanding the code, I think this organization is confusing.